src/components/providers/query-provider.tsx

```diff
@@ -0,0 +1,19 @@
+"use client";
+
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+import { ReactQueryStreamedHydration } from "@tanstack/react-query-next-experimental";
```

Comment:
Hmm, I'm not sure here if you benefit from this module, as still you have `use client` directive. That's a new thing for me as well, I'll try to investigate it